### ISSUE TITLE
Error import page lacks details

### ISSUE THREAD

**Issue by stevage** [CartoDB/cartodb#5778]
A bit of a common theme in some of my reports...

<img width="649" alt="screenshot 2015-09-28 21 47 45" src="https://cloud.githubusercontent.com/assets/678344/10134640/a4d00de2-662a-11e5-8141-f1873dc7bcc0.png">

It would be much better for the user to include some diagnostic and contextual information. 

- what URL was attempted?
- what HTTP code was returned?
- any status message?

(In this case, I think probably the source server has some kind of internal failure, so don't worry about trying to solve this one for me...)

**Comment by javisantana** [comment]
can't be more agree, @gfiorav do you know if the import api returns the real error?

**Comment by gfiorav** [comment]
@javisantana 

AFAIK the importer API only returns an error code.

I believe so based on [this code](https://github.com/CartoDB/cartodb/blob/master/app/controllers/carto/api/imports_controller.rb#L14-L27):

```ruby
def show
  import = DataImportsService.new.process_by_id(params[:id])
  render_404 and return if import.nil?

  data = Carto::Api::DataImportPresenter.new(import).api_public_values
  if import.state == Carto::DataImport::STATE_COMPLETE
    data[:any_table_raster] = import.is_raster?

    decorate_twitter_import_data!(data, import)
    decorate_default_visualization_data!(data, import)
  end

  render json: data
end
```

And [further code for `Carto::Api::DataImportPresenter.new(import).api_public_values`](https://github.com/CartoDB/cartodb/blob/master/app/controllers/carto/api/data_import_presenter.rb#L59-L74):

```ruby
def public_values
  values = Hash[PUBLIC_ATTRIBUTES.map{ |attribute| [attribute, @data_import.send(attribute)] }]
  values.merge!('get_error_text' => get_error_text)
  values.merge!('display_name' => display_name)
  values.merge!('queue_id' => @data_import.id)
  values.merge!(success: @data_import.success) if @data_import.final_state?
  values
end

def get_error_text
  if @data_import.error_code.nil?
    nil
  else
    @data_import.error_code.blank? ? CartoDB::IMPORTER_ERROR_CODES[99999] : CartoDB::IMPORTER_ERROR_CODES[@data_import.error_code]
  end
end
```

where it can be seen that only the result of `get_error_text` is appended and that is an error code.

However, using the identifier, one could get the full error log back, although it's far from UI-friendly (full stack traces).

Should we try to improve this?

**Comment by gfiorav** [comment]
@javisantana @rafatower 

So an error from one part might have different interesting parts than another part (i.e., in this case URL is important but in 'Unsupported/Unrecognized file type' URL is irrelevant). 

Do you have any strong opinions (or suggestion) of how to go about this?

**Comment by rafatower** [comment]
I don't understand what you mean, @gfiorav

**Comment by gfiorav** [comment]
@rafatower 

Sorry if I seemed unclear! :grin: 

What I mean is that if the purpose is to have the importer return more specific data (right now it only returns an error code) we need to specify *what* additional data is returned.

@stevage suggested:
+ URL
+ HTTP code
+ status message

My point is that not all may apply in all cases. For example: 
+ *Remote URL returned an error* -> HTTP code and URL are important.
+ *Filetype not supported* -> None of that is relevant in this case.

My question is: should we follow some kind of schema for deciding what information to show? Should we show all info even if it doesn't apply?

**Comment by rafatower** [comment]
> My question is: should we follow some kind of schema for deciding what information to show?

probably, yes.

> Should we show all info even if it doesn't apply?

I don't think so, what do you think?

**Comment by gfiorav** [comment]
I'm thinking that it might be better to just return more info in all cases and have the frontend decide what to show and when.

We'll need a frontend for this one!

**Comment by rafatower** [comment]
To log what we talked: you can count on Design for review but can carry on with erb template changes.

**Comment by gfiorav** [comment]
So it seems to me this is quite vertical. I have yet to understand where Backbone is loading the models (have 0 experience here).

I think that with a few questions here and there I'll pull it off. If I'm wrong I'll ask for a collaborator!

**Comment by gfiorav** [comment]
Ok, so I piped some more info from `DataImport` model to the front end. Now we must decide what is relevant and what not. 

Also, we need someone to design it. Luckily, @saleiva has volunteered :smile: 

Here's a screenshot:
<img width="745" alt="screen shot 2015-10-01 at 19 36 25" src="https://cloud.githubusercontent.com/assets/4222826/10228606/fd8f4fb0-6873-11e5-917c-055ad85de0bc.png">

**Comment by rafatower** [comment]
I wouldn't expose any internal network information (upload host, pid and so on)

**Comment by gfiorav** [comment]
@rafatower good point. It's already fixed in #5834
